pkg/gameserverallocations/allocator.go

```diff
@@ -607,7 +607,37 @@ func (c *Allocator) applyAllocationToGameServer(ctx context.Context, mp allocati
 	gs.ObjectMeta.Annotations[LastAllocatedAnnotationKey] = string(ts)
 	gs.Status.State = agonesv1.GameServerStateAllocated
 
-	return c.gameServerGetter.GameServers(gs.ObjectMeta.Namespace).Update(ctx, gs, metav1.UpdateOptions{})
+	// perfom any Counter or List actions
+	var counterErrors error
+	var listErrors error
+	if runtime.FeatureEnabled(runtime.FeatureCountsAndLists) {
+		if gsa.Spec.Counters != nil {
+			for counter, ca := range gsa.Spec.Counters {
+				counterErrors = goErrors.Join(counterErrors, ca.CounterActions(counter, gs))
+			}
+		}
+		if gsa.Spec.Lists != nil {
+			for list, la := range gsa.Spec.Lists {
+				listErrors = goErrors.Join(listErrors, la.ListActions(list, gs))
+			}
+		}
+	}
+
+	gsUpdate, updateErr := c.gameServerGetter.GameServers(gs.ObjectMeta.Namespace).Update(ctx, gs, metav1.UpdateOptions{})
+	if updateErr != nil {
+		return gsUpdate, updateErr
+	}
+
+	// If successful Update record any Counter or List action errors as a warning
+	if counterErrors != nil {
+		c.recorder.Event(gsUpdate, corev1.EventTypeWarning, "Counter Errors", counterErrors.Error())
```

Comment:
The reason field of the EventRecorder will be used to automate handling of events and should be UpperCamelCase format. How about "CounterActionError" and "ListActionError"